pkg/util/ipvs/testing/fake_test.go

```diff
@@ -113,46 +113,68 @@ func TestRealServer(t *testing.T) {
 		Port:     uint16(80),
 		Protocol: string("TCP"),
 	}
+	rsInfos := []struct {
+		ip     string
+		port   uint16
+		weight int
+	}{
+		{"172.16.2.1", 8080, 1},
+		{"172.16.2.2", 8080, 2},
+		{"172.16.2.3", 8080, 3},
+	}
+	var rss []*utilipvs.RealServer
 	err := fake.AddVirtualServer(vs)
 	if err != nil {
 		t.Errorf("Fail to add virutal server, error: %v", err)
 	}
-	// Add a real server to the virtual server
-	rs1 := &utilipvs.RealServer{
-		Address: net.ParseIP("172.16.2.1"),
-		Port:    uint16(8080),
-		Weight:  1,
-	}
-	err = fake.AddRealServer(vs, rs1)
-	if err != nil {
-		t.Errorf("Fail to add real server, error: %v", err)
-	}
-	// Add another real server to the virtual server
-	rs2 := &utilipvs.RealServer{
-		Address: net.ParseIP("172.16.3.2"),
-		Port:    uint16(8080),
-		Weight:  2,
-	}
-	err = fake.AddRealServer(vs, rs2)
-	if err != nil {
-		t.Errorf("Fail to add real server, error: %v", err)
+	// Add real server to the virtual server
+	for i, rsInfo := range rsInfos {
+		rss = append(rss, &utilipvs.RealServer{
+			Address: net.ParseIP(rsInfo.ip),
+			Port:    rsInfo.port,
+			Weight:  rsInfo.weight,
+		})
+		if err = fake.AddRealServer(vs, rss[i]); err != nil {
+			t.Errorf("Fail to add real server, error: %v", err)
+		}
 	}
 	// List all real servers of the virtual server
 	list, err := fake.GetRealServers(vs)
 	if err != nil {
 		t.Errorf("Fail to get real servers of the virtual server, error: %v", err)
 	}
-	if len(list) != 2 {
-		t.Errorf("Expect 2 virutal servers, got: %d", len(list))
+	if len(list) != len(rss) {
+		t.Errorf("Expect %d virutal servers, got: %d", len(rss), len(list))
 	}
 	// Delete a real server of the virtual server
-	err = fake.DeleteRealServer(vs, rs2)
-	list, err = fake.GetRealServers(vs)
-	if err != nil {
-		t.Errorf("Fail to get real servers of the virtual server, error: %v", err)
+	// Make sure any position of the list can be real deleted
+	rssLen := len(rss)
+	for i := range rss {
+		rs := rss[rssLen-i-1]
+		if err = fake.DeleteRealServer(vs, rs); err != nil {
+			t.Errorf("Fail to delete real server(%d/%d) of the virtual server, error: %v", rssLen-i, rssLen, err)
+		} else {
```

Comment:
I Can't understand what we are doing here?

Reply:
I want test delete real server in any position of the list, but if I delete it in `rss` order, will always delete the first one in the destination list...So I use the opposite order,,,

Reply:
> I want test delete real server in any position of the list

NO need indeed. The logic is too complicated for test and seems over-killed now. 

My question is:

what are we doing here if failed to delete real server?

Reply:
ehhhh, I see...
It's print total real server count and the failed deleted real server number,,,It is a little confused...

Reply:
If just want to log the error, it's really too complicated.

Reply:
LOL